### PR TITLE
Generalizes some reagent heating

### LINKED ISSUE #29557: Spray bottle heating is bugged
I was testing out multiple liquid containers and noticed that spray bottles don't heat up from lighters, igniters, etc.

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
fix: Fixed hot items and fire heating reagents to temperatures higher than their heat source.
fix: Sprays can now be heated with hot items.
tweak: The heating of reagents by hot items, fire, and microwaves now scales linearly with the difference between the reagent holder's temperature and the heat source's temperature instead of constantly.
tweak: The body temperature of a mob with reagents is affected by the temperature of the mob's reagents.
tweak: Reagent containers can now be heated by hot air.
/:cl:

Fixes https://github.com/tgstation/tgstation/issues/29557 and https://github.com/tgstation/tgstation/issues/32372
 
Added expose_temperature proc to be used for heating reagents in a consistent manner instead of different reagent holders being heated differently. It heats linearly similar to the way the chem heater works.
Added temperature_expose for reagent containers so that they can be heated by the environment for realism or something.
Metabolizing reagents causes the temperatures of the reagents to quickly, but not instantly, change towards the body temperature. Instantly changing to the body temperature prevented expected behavior of some reagents and reactions (i.e. cryostylane synthesis) from occurring.

### CODE REVIEW

**Comment by ACCount12** [comment]
Will heating the mob still heat up the reagents?

**Comment by swindly** [comment]
Heating the mob still heats up the reagents.

**Comment by swindly** [comment]
Changed metabolizing temperature change so that it quickly changes to meet the body temperature. Changing the body temperature towards the reagent temperature allowed high reagent volumes to prevent the body temperature from reaching extremes.

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -230,7 +230,7 @@
 	var/list/cached_reagents = reagent_list
 	var/list/cached_addictions = addiction_list
 	if(C)
-		chem_temp = C.bodytemperature
+		expose_temperature(C.bodytemperature, 0.25)
 		handle_reactions()
```
Comment (AnturK) [inline code/modules/reagents/chemistry/holder.dm]: Not needed anymore ?

File: code/modules/reagents/reagent_containers.dm [hunk 2]
```diff
@@ -125,6 +124,9 @@
 
 /obj/item/reagent_containers/microwave_act(obj/machinery/microwave/M)
 	if(is_open_container())
-		reagents.chem_temp = max(reagents.chem_temp, 1000)
+		reagents.expose_temperature(1000)
 		reagents.handle_reactions()
```
Comment (AnturK) [inline code/modules/reagents/reagent_containers.dm]: Same here ?

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**